consensus/istanbul/backend/backend.go

```diff
@@ -316,3 +352,64 @@ func (sb *backend) HasBadProposal(hash common.Hash) bool {
 func (sb *backend) Close() error {
 	return nil
 }
+
+// IsQBFTConsensus returns whether qbft consensus should be used
+func (sb *backend) IsQBFTConsensus() bool {
+	// If qbftBlock is not defined in genesis, then use legacy ibft
+	if sb.qbftConsensusEnabled {
+		return true
+	}
+	if sb.chain != nil {
+		return sb.IsQBFTConsensusForHeader(sb.chain.CurrentHeader())
+	}
+	return false
+}
+
+// IsQBFTConsensusForHeader checks if qbft consensus is enabled for the block height identified by the given header
+func (sb *backend) IsQBFTConsensusForHeader(header *types.Header) bool {
+	// If qbftBlock is not defined in genesis, then use legacy ibft
+	if sb.config.QbftBlock == nil {
+		return false
+	}
+	if sb.config.QbftBlock.Uint64() == 0 {
+		return true
+	}
+
+	if sb.chain != nil && header.Number.Cmp(sb.config.QbftBlock) >= 0 {
+		return true
+	}
+	return false
+}
+
+// qbftBlockNumber returns the qbftBlock fork block number
+func (sb *backend) qbftBlockNumber() uint64 {
+	if sb.config.QbftBlock == nil {
+		return 0
+	}
```

Comment:
`sb.config.QbftBlock == nil` probably should not result in forkBlock being returned `0` as `nil` config implies qbft is disabled.  Callers may not expect this behaviour.

Reply:
Returning `-1` when `QbftBlock` is not defined